### PR TITLE
Improve README with code examples for logout

### PR DESCRIPTION
**Pull request by josecolella**
Add code examples for handling IDP initiated logout, using authenticate middleware, and code example for SP initiated logout.

@markstos

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,6 +163,21 @@ app.post('/login/callback',
 );
 ```
 
+- The authenticate middleware is also used for the LogoutResponse that will be sent when IDP-initiates a logout as a result of a user logging out of an application. 
+*This will generate a LogoutResponse in response to an IDP LogoutRequest, effectively logging the user out of the application, as well as any other SAML enabled application*
+
+```javascript
+const bodyParser = require('body-parser');
+
+app.post('/saml/logout/callback',
```
[the remark this answers is not in the document]
Comment (josecolella) [reply README.md]: Correct

File: README.md [hunk 2]
```diff
@@ -187,6 +202,33 @@ app.get('/login',
...
+        if (err) {
+            logger.error(err);
+        }
+        req.session.destroy((error) => {
+            if (error) {
+                logger.error(error);
+            }
+            req.logout();
+            res.redirect(uri);
+        });
```
[the remark this answers is not in the document]
Comment (josecolella) [reply README.md]: I agree.

**Comment by josecolella** [comment]
@markstos I will move all these to a wiki and link then in the README. Documenting these things will save hours for many others that will eventually face these issues.

**Comment by markstos** [comment]
@josecolella Please submit a PR to link to the wiki page when it's ready. I agree the contribution will be a time-saver for others.

**Pull request closed without merging**